### ISSUE TITLE
drop nulls and drop zeros returns incorrect timestamp

### ISSUE THREAD

**Issue by ninelocks** [graphite-project/whisper#250]
Maybe I am missing the intention but if I use --drop nulls the outputted times are wrong as t does not account for the time position of the value in the original list. The result is a sequential list of times with no gaps and the timestamps are wrong.

rather  than using the drop function I using the hacky fix below

```
 if value is None:
     t += step       <===== making sure we account for the removed value
    continue  
```

**Comment by deniszh** [comment]
Hi @ninelocks,
Well, agree - but that's how drop is implemented by its author - https://github.com/graphite-project/whisper/pull/57

**Comment by Geoff99** [comment]
Hi @ninelocks, @deniszh 

For what it's worth, I think the current behaviour is pretty clearly a **bug** in the current implementation.

I've just wasted a whole morning checking my code which processes `whisper-fetch.py --drop nulls …` output.  I suspected I had a buffer problem with a subprocess pipe.  Only **finally** did I discover that the `whisper-fetch.py` `--drop` option simply culls the null (or zero, whatever is specified) entries from the  `values` list, **without** in any way adjusting the calculation of `t` variable, (which is formatted into the `timestr` variable) in the final output `for` loop.  

`t` is just incremented by `step` after each iteration of the `for` loop, which means that all `timestr` values will become progressively more incorrect, each time a null or zero value is dropped.
